hazelcast-utils/src/main/java/com/hazelcast/logging/ILogger.java

```diff
@@ -195,15 +193,6 @@ public interface ILogger extends TpcLogger {
      */
     void log(Level level, String message, Throwable thrown);
 
-    /**
-     * Logs a LogEvent.
-     *
-     * @param logEvent the logEvent to log
-     * @deprecated Since 5.1, the method is unused
-     */
-    @Deprecated
-    void log(LogEvent logEvent);
```

Comment:
This is a breaking change for users who either use this method to log, or who implemented their own logging provider.

We either follow semver and don't do this in 5.4, but in 6.0, or we need to announce that we no longer follow semver.

Reply:
The problem is that then this modularization is impossible because through the LogEvent, there is a dependency on Member.

```
public class LogEvent extends EventObject {

    private final LogRecord logRecord;
    private final Member member;

    public LogEvent(LogRecord logRecord, Member member) {
        super(member);
        this.logRecord = logRecord;
        this.member = member;
    }

    public Member getMember() {
        return member;
    }

    public LogRecord getLogRecord() {
        return logRecord;
    }
}
```

Reply:
Perhaps we can find a bytecode rewriting solution. If we can just get a bridge method + bridge constructor, we would be good.

Reply:
I went for a different approach for now. I created an empty com.hazelcast.cluster.Member interface which is in the same location as the main module and the hazelcast-utils compiles against this interface.

In the final jar, the com.hazelcast.cluster.Member of the main hazelcast module overwrites the one from the hazelcast-utils module.

This way we do not break any code but we can start moving the logging functionality to the utils module and we open the door for further modularization.

Reply:
This should not be an issue any longer with the latest change.